Formula/spotbugs.rb

```diff
@@ -13,18 +12,19 @@ class Spotbugs < Formula
 
   bottle :unneeded
 
-  depends_on "openjdk@11"
+  depends_on "openjdk"
 
   def install
-    ENV["JAVA_HOME"] = Formula["openjdk@11"].opt_prefix
+    ENV["JAVA_HOME"] = Formula["openjdk"].opt_prefix
     if build.head?
       system "gradle", "build"
       system "gradle", "installDist"
       libexec.install Dir["spotbugs/build/install/spotbugs/*"]
     else
       libexec.install Dir["*"]
+      chmod 0755, "#{libexec}/bin/spotbugs"
     end
-    (bin/"spotbugs").write_env_script "#{libexec}/bin/spotbugs", :JAVA_HOME => "${JAVA_HOME:-#{ENV["JAVA_HOME"]}}"
+    (bin/"spotbugs").write_env_script "#{libexec}/bin/spotbugs", :JAVA_HOME => Formula["openjdk"].opt_prefix
```

Comment:
Is there a particular reason why setting `JAVA_HOME` from outside is not allowed anymore?
Or we could revert it back?

Reply:
Nope no technical reason it just seemed to be the more common way:
```bash
$ rg -cF 'Formula["openjdk"].opt_prefix' | wc -l
    106
$ rg -cF '${JAVA_HOME:-#{ENV["JAVA_HOME"]}}' | wc -l
    2
```

Reply:
Ah, I see.
It because usually we write something like 
```ruby
"${JAVA_HOME:-#{Formula["openjdk"].opt_prefix}}"
```

But for this case, we already have `ENV["JAVA_HOME"] = Formula["openjdk"].opt_prefix` already, we have it.

Feel free to revert it back of change to 
```suggestion
    (bin/"spotbugs").write_env_script "#{libexec}/bin/spotbugs", :JAVA_HOME => "${JAVA_HOME:-#{Formula["openjdk"].opt_prefix}}"
```

Reply:
fixed it and squashed the commits